### PR TITLE
Add CsvWriter and Storage classes

### LINKED ISSUE #22: Implement Storage class
Implement code for Storage class to handle the loading and saving of the data file.

Satisfies User Story #8

### LINKED ISSUE #24: Implement CsvWriter class for Storage
Implement CsvWriter class for Storage to convert from Collections to data file.

Satisfies User Story #8

### PR DESCRIPTION
**Pull request by NgYaoDong**
- Add JUnit tests for all Storage classes as well.
- Add logging for Storage class.

Fixes #22 and #24.

### CODE REVIEW

**Comment by leongxingyu** [comment]
LGTM.

**Review by leongxingyu — COMMENTED** [review]
LGTM

File: src/main/java/seedu/stockpal/data/product/Description.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package seedu.stockpal.data.product;
 
 public class Description {
+    private static final String EMPTY_STRING = "";
```
Comment (Kobot7) [inline src/main/java/seedu/stockpal/data/product/Description.java:4]: Perhaps shift this to Common.java as it is used in multiple classes

**Review by EdmundTangg — COMMENTED** [review]
LGTM.

**Pull request merged**